src/Packetery/Module/Shipping/ShippingProvider.php

```diff
@@ -229,8 +229,10 @@ public function getSortedCachedMethods( array $originalMethods ): array {
 			return $originalMethods;
 		}
 
+		$zoneId = $this->contextResolver->getShippingZoneId();
+
 		// phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
-		$cacheKey = crc32( serialize( $originalMethods ) );
+		$cacheKey = crc32( serialize( $originalMethods ) . '_zone_' . ( $zoneId ?? 'null' ) );
 
 		if ( ! isset( self::$sortedMethodsCache[ $cacheKey ] ) ) {
 			$unsortedMethods                       = $this->addMethods( $originalMethods );
```

Comment:
Method `getShippingZoneId` is later called again in method `addMethods`. I would pass the variable `$zoneId` to the `addMethods` method to get rid of the duplicity. This could also fix phpunit tests, which have to pass in order to merge the pull request.